logger/logger.go

```diff
@@ -0,0 +1,48 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+package logger
+
+import (
+	"io"
+	"sync"
+
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+)
+
+var (
+	initializeOnce    sync.Once
+	singletonInstance *CWALogger
+)
+
+type CWALogger struct {
```

Comment:
This whole struct really just exists to convert a `Writer` and a `Level` into a `[]zap.Option`... couldn't `getLoggingOptions` just be a helper method in amazon-cloudwatch-agent.go?  The `sync.Once` thing isn't even doing anything for you since it's only called once at startup anyway.

Reply:
I will clean this up a bit